src/java.base/share/classes/java/util/zip/ZipEntry.java

```diff
@@ -91,7 +91,11 @@ public class ZipEntry implements ZipConstants, Cloneable {
      */
     private static final long UPPER_DOSTIME_BOUND =
             128L * 365 * 24 * 60 * 60 * 1000;
-
+    // Maximum possible size of name length + comment length + extra length
+    // for entries in order to not exceed 65,489 bytes minus 46 bytes for the CEN
+    // header length
+    private static final int MAX_NAME_COMMENT_EXTRA_SIZE =
```

Comment:
Would it make sense to validate the actual combined length here, instead of just pretending that all other components of the combined lenghts are zero-length?

We could introduce a method like this:

```java
/*
 * Validate that the combined length of the 46 byte CEN header,
 * the extra field and the comment does not exceed 0xFFFF
 */
private void validateCombinedLength(String name, byte[] extra, String comment, String message) {
    int headerLength = ZipEntry.CENHDR;

    if (name != null) {
        headerLength += name.length();
    }
    if (extra != null) {
        headerLength += extra.length;
    }

    if (comment != null) {
        headerLength += comment.length();
    }

    if (headerLength > 0xFFFF) {
        throw new IllegalArgumentException(message);
    }
}
```

Then the constructor could use it like this:
```java
validateCombinedLength(name, null, null, "entry name too long");
```

setExtra:

```java
validateCombinedLength(name, extra, comment, "invalid extra field length");
```

setComment:
```java
validateCombinedLength(name, extra, comment, "entry comment too long");
```

Reply:
I had thought about that and decided to keep the changes as they are.  I am not opposed to revisiting this in a follow on PR.  Any additional changes would require more javadoc updates to address the overall change in validation. 

So after we fork JDK 24, happyt to revisit.

Reply:
> I had thought about that and decided to keep the changes as they are. I am not opposed to revisiting this in a follow on PR. Any additional changes would require more javadoc updates to address the overall change in validation.
> 
> So after we fork JDK 24, happyt to revisit.

If the route we're taking ends up with having `ZipEntry` manage its own invariant here, then I'm only lukewarm to including this solution in 24 which only takes us half way and has weaker validation than what's already in place in `ZipOutputStream`.  There would be less API churn if we hold our breath here and do it "right" in a single release.

But that's my subjective opinion, it's understandable and fine that others see it differently.

Reply:
> But that's my subjective opinion, it's understandable and fine that others see it differently.

Again, I understand your suggestion and will give it some  additional thought. The original intent was to address the incorrect max value that each of the 3 fields were being validated against, which has been there since at least JDK 1.3.
 
Overall this is a corner case and out of a search of 90,000+ jars, only 520 CEN Headers were encountered with a size between 500-1000 bytes, all other entries were < 500 bytes.

Reply:
Ok,  I revised ZipEntry to incorporate your suggestion.   I kept the validation in ZipOutputStream::putNextEntry in the unlikely event ZipEntry is extended based on a suggestion from Jai.